code/lib/core-server/src/utils/StoryIndexGenerator.test.ts

```diff
@@ -0,0 +1,1741 @@
+                        "title": "A",
+                        "type": "story",
+                      },
+                    },
+                    "v": 4,
+                  }
+              `);
+      });
+
+      it('auto-generates id from name and title inputs', async () => {
+        const specifier: NormalizedStoriesSpecifier = normalizeStoriesEntry(
+          './src/A.stories.js',
+          options
+        );
+
+        const generator = new StoryIndexGenerator([specifier], {
+          ...options,
+          indexers: [
+            {
+              test: /\.stories\.(m?js|ts)x?$/,
+              index: async (fileName) => [
+                {
+                  key: 'StoryOne',
+                  name: 'Story One',
+                  title: 'A',
+                  tags: ['story-tag-from-indexer'],
+                  importPath: fileName,
+                  type: 'story',
+                },
+              ],
+            },
+          ],
+        });
+        await generator.initialize();
+
+        expect(await generator.getIndex()).toMatchInlineSnapshot(`
+          Object {
+            "entries": Object {
+              "a--story-one": Object {
+                "id": "a--story-one",
+                "importPath": "./src/A.stories.js",
+                "name": "Story One",
+                "tags": Array [
+                  "story-tag-from-indexer",
+                  "story",
+                ],
+                "title": "A",
+                "type": "story",
+              },
+            },
+            "v": 4,
+          }
+        `);
+      });
+
+      it('auto-generates id, title and name from key input', async () => {
+        const specifier: NormalizedStoriesSpecifier = normalizeStoriesEntry(
+          './src/A.stories.js',
+          options
+        );
+
+        const generator = new StoryIndexGenerator([specifier], {
+          ...options,
+          indexers: [
+            {
+              test: /\.stories\.(m?js|ts)x?$/,
+              index: async (fileName) => [
+                {
+                  key: 'StoryOne',
+                  tags: ['story-tag-from-indexer'],
+                  importPath: fileName,
+                  type: 'story',
+                },
+              ],
+            },
+          ],
+        });
+        await generator.initialize();
+
+        expect(await generator.getIndex()).toMatchInlineSnapshot(`
+          Object {
+            "entries": Object {
+              "a--story-one": Object {
+                "id": "a--story-one",
+                "importPath": "./src/A.stories.js",
+                "name": "Story One",
+                "tags": Array [
+                  "story-tag-from-indexer",
+                  "story",
+                ],
+                "title": "A",
+                "type": "story",
+              },
+            },
+            "v": 4,
+          }
+        `);
+      });
+    });
+
+    describe('single file specifier', () => {
```

Comment:
everything below is just the existing tests, without any modifications.

Reply:
I wonder if we should split this into more of a "unit test" file (that checks the mapping from `IndexInput` -> `IndexEntry`) test file and then more of an "integration test" file (more or less the existing tests).

Reply:
Agree, I'll do that.